server/channels/app/platform/license.go

```diff
@@ -126,11 +126,19 @@ func (ps *PlatformService) SaveLicense(licenseBytes []byte) (*model.License, *mo
 		return nil, model.NewAppError("addLicense", "api.unmarshal_error", nil, "", http.StatusInternalServerError).Wrap(jsonErr)
 	}
 
+	if license.Features == nil {
+		return nil, model.NewAppError("addLicense", "api.license.add_license.invalid.app_error", nil, "", http.StatusBadRequest)
+	}
+
 	uniqueUserCount, err := ps.Store.User().Count(model.UserCountOptions{})
 	if err != nil {
 		return nil, model.NewAppError("addLicense", "api.license.add_license.invalid_count.app_error", nil, "", http.StatusBadRequest).Wrap(err)
 	}
 
+	if license.Features.Users == nil {
+		return nil, model.NewAppError("addLicense", "api.license.add_license.invalid.app_error", nil, "", http.StatusBadRequest)
```

Comment:
Since we are using the same error string, some context on which part of license is invalid might be helpful.

Reply:
I first added different new errors, but I don't know if that's something the admins can act on. What about something like this?
```
model.NewAppError("addLicense", "api.license.add_license.invalid.app_error", nil, "", http.StatusBadRequest).Wrap(fmt.Errorf("license.Features.Users is nil"))
```

Reply:
Yep that works. Although since there's no formatting, you could simply use `errors.New` instead of `fmt.Errorf`.

Reply:
Done in https://github.com/mattermost/mattermost/pull/29417/commits/c3df02189cd6e031bb68b0d90bc00fd94ea2f6aa